IPython/core/oinspect.py

```diff
@@ -416,6 +416,8 @@ def _getdef(self,obj,oname='') -> Union[str,None]:
 
         If any exception is generated, None is returned instead and the
         exception is suppressed."""
+        if not callable(obj):
+            return None
         try:
             return _render_signature(signature(obj), oname)
```

Comment:
the implementation of signature(obj) starts with:

```
    if not callable(obj):
        raise TypeError('{!r} is not a callable object'.format(obj))
```

The current implementation catches this exception and returns None.
constructing the exception error message can be very expensive (minutes) depending on the obj instance.